### ISSUE TITLE
The more I play the worse it gets

### ISSUE THREAD

**Issue by Astrepio** [SmartlyDressedGames/Unturned-3.x-Community#4177]
I ran into a problem when playing Unturned. Everything is good and stable for 20-30 minutes. But after my FPS begins to drop significantly and drops to 80, although it is stable at 160. I don’t understand why it is so. I tried changing the graphics and option settings in the game - nothing helped. I checked the integrity of the files in Steam and didn't help either. I'm guessing that over time my game gets some kind of cache full or something, I don't know.

Maybe useless info: when i press T (changing modules) my fps drops to 20 for 2 secs and back to 80. Same thing for opening my inventory and pressing ESC.

What should I do to fix it?

**Comment by GazziFX** [comment]
Can you try to monitor memory usage? It might be freezing when you out of memory and system swap to pagefile

**Comment by SDGNelson** [comment]
GazziFX might be right. Worth checking the memory usage?

**Comment by Astrepio** [comment]
I guess 50% is not that much 
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/144468384/e9ba179a-4ee0-4a87-8cff-891289d267fc)

**Comment by mkspeed0** [comment]
You've mentioned that it starts out stable and then drops after 20-30 minutes, which could indicate a performance issue. Here are some steps you can take to try and resolve this problem:

1. Check Your Hardware Temperature: Overheating can cause performance drops. Ensure that your CPU and GPU are not overheating. Use monitoring software like MSI Afterburner or HWMonitor to check your temperatures while playing.

2. Update Graphics Drivers: Make sure you have the latest graphics drivers for your GPU installed. Outdated drivers can cause performance issues. Visit the website of your GPU manufacturer (NVIDIA, AMD, or Intel) to download and install the latest drivers.

3. Lower Graphics Settings: You mentioned that you've already tried changing graphics settings, but make sure you've set them to the lowest possible settings to see if the problem persists. If performance improves, gradually increase settings until you find a balance between graphics quality and performance.

4. Background Processes: Close any unnecessary background processes and applications that might be using system resources while you're playing the game. This includes browser tabs, video players, and other resource-intensive applications.

5. Monitor System Resources: Use a program like Task Manager (Ctrl+Shift+Esc) to monitor your CPU, RAM, and disk usage while playing the game. Look for any processes that are using an unusually high amount of resources.

6. Verify Game Files: You mentioned that you've already done this, but it's a good practice to verify game files through Steam to ensure there are no corrupted files. You can also try reinstalling the game if the issue persists.

7. Check for Software Conflicts: Sometimes, background software like antivirus or overlay applications can interfere with the game's performance. Try disabling or closing such software while playing.

8. Update Windows: Ensure that your operating system is up to date with the latest Windows updates, as these can sometimes include important performance optimizations.

9. Check for Disk Fragmentation: Over time, your hard drive can become fragmented, which can affect game loading times. Use Windows' built-in tool to defragment your hard drive.

10. Clear Game Cache: Some games create temporary cache files that can accumulate over time. Check if Unturned has an option to clear its cache, and if not, you might try uninstalling and reinstalling the game.

11. Check for Memory Leaks: Memory leaks in a game or in the game engine can cause performance degradation over time. Unfortunately, you may not be able to fix this yourself; the game developer would need to release a patch.

12. Monitor Network Traffic: Sometimes, online games can be affected by network issues. Use a network monitoring tool to check for any unusual spikes or drops in your network traffic.

**Comment by Astrepio** [comment]
Ok, I will try to do it. Thank you

**Comment by SDGNelson** [comment]
I received a support ticket from another player which seemed like a similar situation. Nothing jumps to mind as potentially causing gradually degrading performance, but I will see if I can reproduce it. Is there a particular server you were on that I should try joining?

**Comment by Astrepio** [comment]
@SDGNelson nah, you don't have to join any servers. This issue happens in any server after 40 minutes of gameplay. I did a lot of things to fix it (updated graphic drivers, reinstalled the game and so on). So idk but i really want to play with the best experience.

**Comment by Astrepio** [comment]
@SDGNelson this issue has changed a bit. I dont know why but now i dont have to restart the game in order to fix it. Just to rejoin a server and i will have normal fps for 40 min and then to rejoin server again.

**Comment by GazziFX** [comment]
Did you experience this problem before the 3.23.12.0 update?

**Comment by Astrepio** [comment]
> Did you experience this problem before the 3.23.12.0 update?

Yes

**Comment by ItsRodrigoAl** [comment]
I can confirm but not in the same way. I have a 3GB GTX 1060 so I can play at medium/high graphics at 80-120fps (I have limited it at 90fps)

Basically I left my game running for days, after some days even if it was marking 90fps it was like running at 40fps

**Comment by SDGNelson** [comment]
Sorry, I forgot to follow up on this:

Earlier this week I ran a debug build playing in singleplayer for ~8 hours. Each time the character died I ran around again to activate some zombie activity but otherwise was idle. Unfortunately, no significant performance change.

The next day I did the same thing on a local multiplayer server with the same results.

That being said, maybe the problem was using too high specs. It sounds like in ItsRodrigoAl's case it was over the course of multiple days in-game? @Astrepio do you leave the game running for long periods of time or is it within a short period of startup? And can you share your PC specs please - I have a variety of PC hardware from the the span of the past 10 years so maybe I can try using whatever I have that's closest.

**Comment by DanielWillett** [comment]
Have you made sure that you're using the right graphics card? 
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/12886600/faa4930c-54e9-4083-a8ac-6ef91b8a7c61)

**Comment by Astrepio** [comment]
Well, it was auto-select but now i have changed it. I will check and answer later

**Comment by Astrepio** [comment]
Didn't help me

**Comment by BoomViz** [comment]
Actually, I've encountered this issue myself—I even posted about it on the forum at one point.
https://forum.smartlydressedgames.com/t/chat-history-length/24501

I thought the problem was with my old PC, but after upgrading, I realized that even on a powerful new rig, I experienced massive FPS drops after some time in-game. The whole issue turned out to be the History_Length parameter, which I had set to an extremely high value. Once the chat log filled up, it tanked performance hard. So now I keep my chat history at 50, which is more than enough.

I think you should either limit this setting or add a warning that it can severely impact FPS. Newcomers who see the preferences file might set it to a huge value without realizing the consequences, then wonder why their game starts lagging badly after a while.

**Comment by BoomViz** [comment]
As an example, I wrote a plugin that spammed 5000 messages
https://youtu.be/9Ar8PVVuJkY

**Comment by SDGNelson** [comment]
Holy moly! Yes, we should clamp that option. 😱
